hail/python/hailtop/hailctl/batch/list_batches.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+import tabulate
+
+from .batch_cli_utils import bool_string_to_bool
+
+def init_parser(parser):
+    parser.add_argument('--success', '-s', type=str, help="true or false")
+    parser.add_argument('--complete', '-c', type=str, help="true or false")
+    parser.add_argument('--attributes', '-a', metavar="KEY=VALUE", nargs='+',
+                        help="Filters list to specified attributes. Specify attributes using"
+                        "KEY=VALUE form, do not put spaces before or after the equal sign.")
+
+def main(args, passthrough_args, client):
+
+    success = None
+    if args.success:
+        try:
+            success = bool_string_to_bool(args.success)
+        except:
+            raise argparse.ArgumentTypeError("Boolean value expected for success")
+
+    complete = None
+    if args.complete:
+        try:
+            complete = bool_string_to_bool(args.complete)
+        except:
+            raise argparse.ArgumentTypeError("Boolean value expected for complete")
+
+    attributes = {}
+    if args.attributes:
+        for att in args.attributes:
+            att = att.strip()
+            key_value = att.split('=')
+            if len(key_value) != 2:
+                raise argparse.ArgumentTypeError('Attribute "{}" should contain exactly one equal sign')
```

Comment:
missing formatter.

I generally like f-strings here--it's very easy to do
```
f'Attribute {att!r} should contain exactly...'
```

The `!r` prints the repr, which will add its own quote.

Reply:
(and properly escape non-printable)

Reply:
Nice didn't know that trick I've been doing `"".format(...)`